sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/FilterAnalyzer.scala

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.models.CosmosParametrizedQuery
+import org.apache.spark.sql.sources.{
+  And, EqualNullSafe, EqualTo, Filter, GreaterThan,
+  GreaterThanOrEqual, In, IsNotNull, IsNull, LessThan, LessThanOrEqual, Not, Or,
+  StringContains, StringEndsWith, StringStartsWith
+}
+
+import scala.collection.mutable.ListBuffer
+
+case class FilterAnalyzer() {
+  def analyze(filters: Array[Filter]): AnalyzedFilters = {
+    val queryBuilder = new StringBuilder
+    queryBuilder.append("SELECT * FROM r")
```

Comment:
In the future, I think this code should leverage a DOM / AST. We have one in .net:

https://github.com/Azure/azure-cosmos-dotnet-v3/tree/master/Microsoft.Azure.Cosmos/src/SqlObjects

and maybe we should port it to Java, since spark connector, seems to be doing some heavy query manipulation.

Reply:
Thanks @bchong95 for the pointer I will take a look at that. Added a TODO in the code for this.